### PR TITLE
CEDS-1726 - refactored remove button hint 

### PR DESCRIPTION
**Pull request by SteveSugden**
to enable sensible remove button messages for item type page

### CODE REVIEW

File: app/views/components/fields/LabelledValue.scala [hunk 1]
```diff
@@ -19,5 +19,5 @@ package views.components.fields
 case class LabelledValue(label: String, value: String)
 
 object LabelledValue {
-  def apply(labelAndValue: String): LabelledValue = LabelledValue(labelAndValue, labelAndValue)
+  def apply(labelAndValue: Any): LabelledValue = LabelledValue(labelAndValue.toString, labelAndValue.toString)
```
Comment (bambuchaAdm) [inline app/views/components/fields/LabelledValue.scala]: Why any ?

Comment (PatrykRudnicki) [inline app/views/components/fields/LabelledValue.scala]: Why we need to change to `Any`? Is it a possibility to receive something different than `String`?

Comment (SteveSugden) [reply app/views/components/fields/LabelledValue.scala]: Yes.  To save the client calling toString on the Json object

File: app/views/declaration/item_type.scala.html [hunk 2]
```diff
@@ -71,7 +71,8 @@
             elements = taricAdditionalCodes,
             labelClass = Some("bold-small"),
             valuePrefix = Some(taricAdditionalCodesKey+"_"),
-            addHint = "declaration.itemType.taricAdditionalCodes.add.hint"
+            addHint = "declaration.itemType.taricAdditionalCodes.add.hint",
+            removeHint = "declaration.itemType.taricAdditionalCodes.remove.hint"
```
Comment (ed0906) [inline app/views/declaration/item_type.scala.html]: 👍

File: app/views/declaration/item_type.scala.html [hunk 3]
```diff
@@ -86,7 +87,8 @@
                 items = AutoCompleteItem.fromNationalAdditionalCode(NationalAdditionalCode.all)
             ),
             valuePrefix = Some(nationalAdditionalCodesKey+"_"),
-            addHint = "declaration.itemType.nationalAdditionalCode.add.hint"
+            addHint = "declaration.itemType.nationalAdditionalCode.add.hint",
+            removeHint = "declaration.itemType.nationalAdditionalCode.remove.hint"
```
Comment (ed0906) [inline app/views/declaration/item_type.scala.html]: 👍

File: app/views/components/fields/LabelledValue.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/views/components/fields/LabelledValue.scala]: Can we provide two apply methods? One for `String` and second one for `JsObject`?

Using `Any` is not a good practice, you can pass everything there.

File: app/views/components/fields/LabelledValue.scala [hunk 4]
```diff
@@ -16,8 +16,11 @@
 
 package views.components.fields
 
+import play.api.libs.json.JsValue
+
 case class LabelledValue(label: String, value: String)
 
 object LabelledValue {
   def apply(labelAndValue: String): LabelledValue = LabelledValue(labelAndValue, labelAndValue)
+  def apply(labelAndValue: JsValue): LabelledValue = LabelledValue(labelAndValue.toString, labelAndValue.toString)
```
Comment (PatrykRudnicki) [inline app/views/components/fields/LabelledValue.scala]: Yeah, it's better now. I'm just wondering why we need to have the same `label` and `value`?

Of course you can create now case class with two arguments, so it's okay, but I'm just curious what case is covered by it.

Comment (bambuchaAdm) [reply app/views/components/fields/LabelledValue.scala]: Two apply is good enough - you have still strongly types and everything should work perfectly.

Comment (bambuchaAdm) [reply app/views/components/fields/LabelledValue.scala]: @PatrykRudnicki this depends on how MultipleItemsHelper  works - it compare sent value with values in model

**Pull request merged**